src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getExtraAttributes
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.Objects
+import kotlin.math.absoluteValue
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object ItemPickupLog {
+
+    enum class DisplayLayout(private val display: String) {
+        ICON("§e✎"),
+        ITEM_NAME("§d[Cute] Skirtwearer's Cake Soul"),
+        CHANGE_AMOUNT("§a+256"),
+        ;
+
+        override fun toString() = display
+    }
+
+    private val config get() = SkyHanniMod.feature.inventory.itemPickupLogConfig
+    private val coinIcon = "COIN_TALISMAN".asInternalName()
+    private var itemList = mutableMapOf<Int, Pair<ItemStack, Int>>()
+    private var itemsAddedToInventory = mutableMapOf<Int, UpdatedItem>()
+    private var itemsRemovedFromInventory = mutableMapOf<Int, UpdatedItem>()
+    private var display: Renderable = Renderable.string("")
+
+    private val patternGroup = RepoPattern.group("itempickuplog")
+    private val shopPattern by patternGroup.pattern(
+        "shoppattern",
+        "^(?<itemName>.+?)(?: x\\d+)?\$",
+    )
+
+    private val bannedItemsPattern by patternGroup.list(
+        "banneditems",
+        "SKYBLOCK_MENU",
+        "CANCEL_PARKOUR_ITEM",
+        "CANCEL_RACE_ITEM",
+    )
+    private val bannedItemsConverted = bannedItemsPattern.map { it.toString().asInternalName() }
+
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!isEnabled()) return
+        config.pos.renderRenderable(display, posLabel = "Item Pickup Log Display")
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        if (!isEnabled()) return
+        itemList.clear()
+        itemsAddedToInventory.clear()
+        itemsRemovedFromInventory.clear()
+    }
+
+    @SubscribeEvent
+    fun onSackChange(event: SackChangeEvent) {
+        if (!isEnabled() || !config.sack) return
+
+        event.sackChanges.forEach {
+            val itemStack = (it.internalName.getItemStack())
+            val item = UpdatedItem(itemStack.displayName, it.delta.absoluteValue.toLong(), it.internalName)
+
+            updateItem(itemStack.hash(), item, itemStack, it.delta < 0)
+        }
+    }
+
+    @SubscribeEvent
+    fun onPurseChange(event: PurseChangeEvent) {
+        if (!isEnabled() || !config.coins || !worldChangeCooldown()) return
+
+        updateItem(0, UpdatedItem("§6Coins", event.coins.absoluteValue.toLong(), coinIcon), coinIcon.getItemStack(), event.coins < 0)
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
```

Comment:
can we use ItemAddInInventoryEvent here instead?

Reply:
We cant with the current state of the event, since that sends the internal name of the item. This isnt ideal with items that have reforges, stars, recombs,...

Reply:
why do we need to work with those stats? you only pickup items that are "clean"

Reply:
Most of the time yes, but sometimes you wont and this will happen
![image](https://github.com/user-attachments/assets/2fb4e071-3b92-4061-a4a3-3b15dec5f727)

Reply:
as discussed in discord, this is necessary to differentiate between item modifiers. Please add a comment to the tick event why we cant use `ItemAddInInventoryEvent ` here.